source/Test.cpp

```diff
@@ -0,0 +1,124 @@
+/* Test.cpp
+Copyright (c) 2019 by Peter van der Meer
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Test.h"
+
+#include "DataNode.h"
+#include "Files.h"
+#include "GameData.h"
+#include "MainPanel.h"
+#include "Panel.h"
+#include "Planet.h"
+#include "PlanetPanel.h"
+#include "PlayerInfo.h"
+#include "Ship.h"
+#include "System.h"
+#include "TestData.h"
+#include "UI.h"
+
+#include <SDL2/SDL.h>
+
+#include <stdexcept>
+
+using namespace std;
+
+
+
+void Test::Load(const DataNode &node)
+{
+	if(node.Size() < 2)
+	{
+		node.PrintTrace("No name specified for test");
+		return;
+	}
+	// If a test object is "loaded" twice, that is most likely an error (e.g.
+	// due to a plugin containing a test with the same name as the base game
+	// or another plugin). Tests should be globally unique.
+	if(!name.empty())
+	{
+		node.PrintTrace("Duplicate definition of test");
+		return;
+	}
+	name = node.Token(1);
```

Comment:
Should we enforce that a tab character is not a part of the test name, since the tab character is reserved for use as a delimiter by the test runner?

Reply:
Sounds good, but I chose to validate against a list of allowed characters instead. This way we also take into account other characters that could lead to problems.